PhillipsAuctionHouse/seldon
# chore(component): L3-2031 add create component script

**Jira ticket**

[L3-2031](https://phillipsauctions.atlassian.net/browse/L3-2031)

**Summary**

Creates a utility script that developers can use to bootstrap new component development.  This will help cover common best practices like our Typescript definitions, test cases, and consistent default props.

**Change List (describe the changes made to the files)**

- add(.template/*): starter files for a new component
- add(utils): getCommonProps refactored to be reused across components
- add(testUtils): adds a runCommonTests utility

**Acceptance Test (how to verify the PR)**

- Run the `npm run create-component NewComponent`
- The new component should show up in storybook
- Change the background color for that component in the scss file.
- The style should be applied in the storybook
- Run `npm run test` on that new Component, the testcases should complete
- You should be able to import the new component with `npm run build && npm link` and use it in remix

**Regression Test**

- N/A

**Evidence of testing**

```
> @phillips/seldon@1.25.1 create-component
> node --no-warnings --loader ts-node/esm .template/scripts/createComponent.ts NewComponent

Component files copied to /Users/ScottDickerson/Documents/GitHub/seldon/src/components/NewComponent
```

![image](https://github.com/PhillipsAuctionHouse/seldon/assets/6663002/f494df25-ad48-400e-865c-05aa1d6047f5)




**Things to look for during review**


New Components



[L3-2031]: https://phillipsauctions.atlassian.net/browse/L3-2031?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
